src/mds/flock.h

```diff
@@ -29,17 +29,52 @@ inline bool ceph_filelock_owner_equal(ceph_filelock& l, ceph_filelock& r)
   return l.pid == r.pid;
 }
 
-inline bool operator==(ceph_filelock& l, ceph_filelock& r) {
-  return
-    l.length == r.length &&
-    l.type == r.type &&
-    ceph_filelock_owner_equal(l, r);
+inline int ceph_filelock_owner_compare(const ceph_filelock& l, const ceph_filelock& r)
+{
+  if (l.client != r.client)
+    return l.client > r.client ? 1 : -1;
+  if (l.owner != r.owner)
+    return l.owner > r.owner ? 1 : -1;
+  if (l.owner & (1ULL << 63))
```

Comment:
Or just copy the comment from ceph_filelock_owner_equal up above; it confused the heck out of me until I looked at the full file and saw that. :)